### PR TITLE
Terminal: add reset button

### LINKED ISSUE #1618: Terminal needs reset
Once a terminal becomes unusable (logout, invalid character sequences) user needs to log out of cockpit and log back in to recover.
image

How about a reset button to get a terminal with the initial state (same as on login)?

### PR DESCRIPTION
**Pull request by dperpeet**
fixes #1618

### CODE REVIEW

**Comment by stefwalter** [comment]
Code looks good at first glance. Could you add a test case to check-terminal?

**Comment by stefwalter** [comment]
Could you add a screenshot of how this looks right now, if this needsdesign?

**Comment by dperpeet** [comment]
current design
screenshot from 2015-01-16 12 15 03

If we only need the button, it could be on the top left (with added padding?)

If we want to add more options later on, e.g. terminal colors or size, a panel on the left might be nice.

**Comment by andreasn** [comment]
I would like to see the control closer to the terminal itself. This would also allow for the shorter label "Reset" rather than the longer "Reset Terminal". In the cases where the terminal is embedded (like on the docker page, but not super-sure that applies), it would feel more like one unit.

Mockup

I've added the little label "Terminal", a bit of info overload, but it helps balance the reset button a bit.
Options could go under a gear menu of some sorts in the same area in the future.

**Comment by dperpeet** [comment]
@andreasn I modified the style.
Please check html/css for consistency.
I override console.terminal.margin. I believe the terminal is used elsewhere, so I didn't want to change the global style.
screenshot from 2015-01-16 23 25 16

**Comment by dperpeet** [comment]
@andreasn is the ``style="display: none;"`` necessary? [terminal.html](https://github.com/dperpeet/cockpit/blob/3970b9e6c674362e1b94f4c2a62aacef073b5953/pkg/server-systemd/terminal.html#L89)

**Comment by andreasn** [comment]
@dperpeet I am not sure about why display: none; @stefwalter or @mvollmer would know.

**Comment by andreasn** [comment]
instead of style="display: none;" it is possible to use the .hide and .show classes from patternfly/bootstrap
http://getbootstrap.com/css/#helper-classes-show-hide

**Comment by stefwalter** [comment]
@dperpeet Is this ready for review?

**Comment by dperpeet** [comment]
There is no new screenshot, since the view didn't change.
@andreasn I couldn't find the necessary style tags elsewhere. Are the classes ok like this?

**Comment by andreasn** [comment]
Looks good!

File: pkg/server-systemd/terminal.html [hunk 1]
```diff
@@ -71,6 +87,15 @@
 </script>
 </head>
 <body style="display: none;">
-    <div id="terminal" class="console"></div>
+    <div id="terminal_container" class="panel panel-default console-container">
+        <div id="terminal_control" class="panel-heading">
+            <span translatable="yes">
+                Terminal
+            </span>
+            <button id="terminal_reset" class="btn btn-default pull-right" translatable="yes">Reset</button>
```
Comment (stefwalter) [inline pkg/server-systemd/terminal.html]: We generally use dashes in our HTML ids. See the coding style guide.

File: pkg/server-systemd/terminal.html [hunk 2]
```diff
@@ -71,6 +87,15 @@
 </script>
 </head>
 <body style="display: none;">
-    <div id="terminal" class="console"></div>
+    <div id="terminal_container" class="panel panel-default console-container">
```
Comment (stefwalter) [inline pkg/server-systemd/terminal.html]: This id doesn't seem to be referenced anywhere. We should probably remove it.

File: pkg/server-systemd/terminal.html [hunk 3]
```diff
@@ -71,6 +87,15 @@
...
+        <div id="terminal_control" class="panel-heading">
```
Comment (stefwalter) [inline pkg/server-systemd/terminal.html]: Same here.

**Comment by stefwalter** [comment]
Noted a few tiny style fixes. Other than that, looks good.

**Comment by stefwalter** [comment]
Merged. Fixed check-terminal use of underscore instead of dash which caused test failure.

**Pull request closed without merging**